lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -55,7 +55,10 @@ class SqlParser(private val ion: IonSystem) : Parser {
         SIMPLE_PATH
     }
 
-    internal enum class ParseType(val isJoin: Boolean = false) {
+    /** If the property [isTopLevelType] is true, then the parse node of this ParseType will only be valid at top level in the query.
+     *  For example, EXEC, DDL and DML keywords can only be used at the top level in the query.
+     */
+    internal enum class ParseType(val isJoin: Boolean = false, val isTopLevelType: Boolean = false, val isDml: Boolean = false) {
```

Comment:
How is the `isDml` parameter defined? `SET` is a DML op, yet `ParseType.SET` `isDml` is `false`.

Reply:
This is a legit observation. However, the `SET` `parseType` is not used anywhere. We can actually get rid of it.

Reply:
Changed the property `isDml` to `true` for `SET`.